src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -806,26 +804,53 @@ private Task AssignVmAsContributorToStorageAccountAsync(IIdentity managedIdentit
         private Task<IStorageAccount> CreateStorageAccountAsync()
         {
             return Execute(
```

Comment:
How about this:
```
        private async Task<IStorageAccount> SetupStorageAccountAsync()
        {
            var storageAccount = await GetExistingStorageAccountAsync(configuration.StorageAccountName);

            return await Execute(
                $"{(storageAccount != null ? "Setting up existing" : "Creating")} Storage Account: {configuration.StorageAccountName}...",
                async () =>
                {
                    storageAccount ??= await azureClient.StorageAccounts
                        .Define(configuration.StorageAccountName)
                        .WithRegion(configuration.RegionName)
                        .WithExistingResourceGroup(configuration.ResourceGroupName)
                        .WithOnlyHttpsTraffic()
                        .CreateAsync(cts.Token);

                    cts.Token.ThrowIfCancellationRequested();
                    var blobClient = await GetBlobClientAsync(storageAccount);

                    var defaultContainers = new List<string> { WorkflowsContainerName, InputsContainerName, "cromwell-executions", "cromwell-workflow-logs", "outputs", ConfigurationContainerName };
                    await Task.WhenAll(defaultContainers.Select(c => blobClient.GetContainerReference(c).CreateIfNotExistsAsync(cts.Token)));

                    return storageAccount;
                });
        }
```

In that case method CreateNewStorageAccountAsync goes away.

Reply:
Thanks Tony.  I don't think it works, because if configuration.StorageAccountName is not null at this stage.  If the storage account isn't specified, it will be set to a name that is based on MainIdentifierPrefix.  That is to say, this happens earlier

```
                    if (string.IsNullOrWhiteSpace(configuration.StorageAccountName))
                    {
                        configuration.StorageAccountName = SdkContext.RandomResourceName($"{configuration.MainIdentifierPrefix}", 24);
                    }
```